src/neo/Consensus/ConsensusContext.cs

```diff
@@ -283,7 +283,7 @@ public ConsensusPayload MakePrepareRequest()
             Span<byte> buffer = stackalloc byte[sizeof(ulong)];
             random.NextBytes(buffer);
             Block.ConsensusData.Nonce = BitConverter.ToUInt64(buffer);
-            EnsureMaxBlockSize(Blockchain.Singleton.MemPool.GetSortedVerifiedTransactions());
+            EnsureMaxBlockSize(Blockchain.Singleton.MemPool.GetSortedVerifiedTransactions(Snapshot));
```

Comment:
I think not yet

Reply:
The request will wait for 2 conditions (any of them)

- A response of it has already been iterated when iterating the mempool, (it means that it will enter the same block as it).
- There is already a result for that tx waiting in the states, (it means that the response has already entered a previous block)

Reply:
You can see the code here 
https://github.com/neo-project/neo/pull/1555/files#diff-eab33c87be022a91edcd35025f268e73R44-R81

Reply:
I think `IsReady` ensures that we're getting both transactions as a result of `GetSortedVerifiedTransactions()`, that's fine, but what I'm talking about is `EnsureMaxBlockSize` filtering, it doesn't seem to be aware of transaction dependencies and we may lose one of them here.